index.ts

```diff
@@ -369,9 +369,154 @@ const CreateOrgSchema = OrgSchema.pick({
+  }, 'Month must be between 1 and 12'),
+  year: z.string().optional().refine((val) => {
+    if (!val) return true; // Optional field
+    const year = parseInt(val);
+    const currentYear = new Date().getFullYear();
+    return year >= 1900 && year <= currentYear;
+  }, `Year must be between 1900 and ${new Date().getFullYear()}`),
+  // Optional relationship fields
+  caregiverId: z.string().optional(),
+  teacherId: z.string().optional(),
+  // Required site field
+  site: z.string().min(1, 'Site is required').trim(),
+  // Optional organizational fields (cohort OR school+class)
+  cohort: z.string().optional(),
+  school: z.string().optional(),
+  class: z.string().optional(),
+}).refine((data) => {
+  // Child users must have month and year
+  if (data.userType === 'child') {
+    return data.month && data.year;
+  }
+  return true;
+}, {
+  message: 'Child users must have month and year',
+  path: ['month', 'year']
+}).refine((data) => {
+  // Must have either cohort OR (site + school)
+  const hasCohort = data.cohort && data.cohort.trim() !== '';
+  const hasSchool = data.school && data.school.trim() !== '';
+  const hasSite = data.site && data.site.trim() !== '';
+  
+  return hasCohort || (hasSite && hasSchool);
+}, {
+  message: 'Must have either cohort OR (site + school)',
+  path: ['cohort', 'school']
+}).refine((data) => {
+  // If class is provided, school must also be provided
+  if (data.class && data.class.trim() !== '') {
+    return data.school && data.school.trim() !== '';
+  }
+  return true;
+}, {
+  message: 'Class requires school to be specified',
+  path: ['school']
+});
+
+// Schema for LinkUsers CSV validation
+const LinkUsersCsvSchema = z.object({
+  id: z.string().min(1, 'ID is required').trim(),
+  userType: z.enum(['child', 'caregiver', 'teacher'], {
+    message: 'userType must be one of: child, caregiver, teacher'
+  }),
+  uid: z.string().min(1, 'UID is required').trim(),
+  // Optional relationship fields
+  caregiverId: z.string().optional(),
+  teacherId: z.string().optional(),
+});
+
+// Schema for validating CSV file structure (headers)
+const CsvHeadersSchema = z.object({
+  headers: z.array(z.string()),
+  requiredHeaders: z.array(z.string()),
+  optionalHeaders: z.array(z.string()).optional(),
+});
+
+// Validation helper functions
+const validateCsvData = <T>(schema: z.ZodSchema<T>, data: unknown[]): {
+  success: boolean;
+  data?: T[];
+  errors?: Array<{
+    row: number;
+    errors: z.ZodError;
+  }>;
+} => {
+  const results: T[] = [];
+  const errors: Array<{ row: number; errors: z.ZodError }> = [];
+
+  data.forEach((row, index) => {
+    const result = schema.safeParse(row);
+    if (result.success) {
+      results.push(result.data);
+    } else {
+      errors.push({ row: index + 1, errors: result.error });
+    }
+  });
+
+  if (errors.length === 0) {
+    return {
+      success: true,
+      data: results,
+    };
+  } else {
+    return {
+      success: false,
+      errors: errors,
```

Comment:
We should have the return object always be the same structure. So it should always return the success, errors, and data property, even if one of them is empty.